### PR TITLE
Improvements for MSlice manual test instructions

### PR DESCRIPTION
**Pull request by SilkeSchomann**
**Description of work.**

Updated the instructions for MSlice manual testing.

**To test:**

Follow the test instructions and see if they make sense. Also, please look for typos and grammatical errors.

The associated issue is https://github.com/mantidproject/mantid/pull/592 in the MSlice project: https://github.com/mantidproject/mslice/issues/647 

*This does not require release notes* because **it is an addition to the developer documentation for manual testing**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst [hunk 1]
```diff
@@ -71,19 +134,124 @@ Taking Cuts
 #. Click ``Display`` in the ``Slice`` tab without changing the default values
 #. On the slice plot, select ``Interactive Cuts``
 #. Use the cursor to select a rectangular region in the slice plot. A second window with a cut plot should open.
+#. Check that the menu item ``Intensity`` is disabled as well as the item ``Recoil lines`` within the menu item ``Information`` in the new plot window
+#. Check that the ``File`` menu only has one menu item, ``Close``
 #. Change the rectangle by changing its size or dragging it to a different area of the slice plot. The cut plot should update accordingly.
+#. Click on ``Save Cut to Workspace`` and check the ``MD Histo`` tab of the Workspace Manager to verify that the new workspace was added
+#. Click on Flip Integration Axis. The y axis label changes from ``Energy Transfer (meV)`` to :math:`|Q| (\mathrm{\AA}^{-1})` or vice versa, depending on the initial label.
```
Comment (sf1919) [inline dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:141]: I can't find Flip Integration Axis as an option. Is it in a menu somewhere? 

Also there is a formatting issue with the :math: part of this line.

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst [hunk 2]
```diff
@@ -71,19 +134,124 @@ Taking Cuts
...
+
+#. Run the script.
+#. There should be two new windows with a slice plot and a cut plot
+
+.. figure:: ../../../../docs/source/images/output_mslice_script.png
+   :alt: output_mslice_script.png
+   :align: center
+   :width: 80%
+
+#. Repeat the same test by copying the script into the Jupyter QtConsole of the MSlice interface
```
Comment (sf1919) [inline dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:207]: I couldn't find the Jupyter QtConsole so wasn't able to test this. I think maybe some more instructions are needed?

**Comment by sf1919** [comment]
I've made a few comments about the testing document as I've been going through and using it. However I came across a few problems that are issues with MSlice rather than the testing documents. These are as follows:

- In the Default Settings part of the testing I could not change Default Energy Units. No menu appeared when clicking this option or hovering over it so no changes could be made.
- In Taking Cuts: Plotting a Cut I had intermittent problems with the cut plot lines not always displaying. Sometimes they would and sometimes they would appear as below. 

- When I tried to modify the setting in the Plot Options (No. 3 on the second set of instructions) I ticked Show Grid and Logarithmic tick boxes. This caused the following error
```
Traceback (most recent call last):
  File "C:\...\build\scripts\ExternalInterfaces\mslice\src\mslice\mslice\presenters\plot_options_presenter.py", line 94, in get_new_config
    self._model.change_axis_scale(self._xy_config)
  File "C:\...\build\scripts\ExternalInterfaces\mslice\src\mslice\mslice\plotting\plot_window\cut_plot.py", line 157, in change_axis_scale
    min = get_min(xdata, absolute_minimum=0.)
  File "C:\...\build\scripts\ExternalInterfaces\mslice\src\mslice\mslice\plotting\plot_window\cut_plot.py", line 23, in get_min
    mask = np.greater(data, absolute_minimum)
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
- The first time I went through all the instructions and got to no.7 in the second list of Plotting a Cut I got very confused. I had a lot more than 2 entries and couldn't find the one for the -1 to 1 cut. However I have had to try and run through these tests a few times and it hasn't happened again so it does not seem to be a consistent problem.
- I could not do the instructions for Generating a Script at all. The ```File Menu``` on the cut plot only had one option, ```Close```!

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst [hunk 3]
```diff
@@ -27,27 +37,58 @@ Taking Slices
...
 
-#. Navigate to the ``File`` menu on the slice plot
-#. Select ``Generate Script to Clipboard`` and paste the script into the Mantid editor. Please note that on Linux ``Ctrl + V`` might not work as expected. Use ``shift insert`` instead in this case.
-#. Run the script and check that the same slice plot is displayed
+#. Navigate to the ``Plots`` tab of MSlice and check that there are entries for two plots
+#. Open the ``Plots`` tab of Mantid and check that there are no entries for plots
+#. Select one of the plots in the ``Plots`` tab of MSlice and click on ``Hide``, the corresponding plot should disappear
+#. Now click on ``Show`` for this plot and it should re-appear again
+#. Double-click on elements of the original slice plot and modify settings, for instance the plot itself and the colorbar axes
+#. Change the plot title and the y axis label to LaTeX, for instance ``$\mathrm{\AA}^{-1}$``, and ensure the text is displayed correctly (for ``$\mathrm{\AA}^{-1}$`` it should be :math:`\mathrm{\AA}^{-1}`)
```
[the remark this answers is not in the document]
Comment (SilkeSchomann) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:60]: This is probably an issue with the build as it displays correctly. Is DOCS_MATH_EXT in CMake set to sphinx.ext.imgmath?

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst (hunk 1 above)
Comment (SilkeSchomann) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:141]: I think the formatting issue is caused by a build setting

**Comment by SilkeSchomann** [comment]
There was a bug impacting Default Energy Units. It is fixed in MSlice but not yet in Mantid.

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (sf1919) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:66]: Yes it is. Does it need to be set as something else?

[the remark this answers is not in the document]
Comment (SilkeSchomann) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:66]: No, it is the correct setting.

[the remark this answers is not in the document]
Comment (sf1919) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:66]: So with that setting and building dev-docs from scratch I get the following:
Screenshot (148)

[the remark this answers is not in the document]
Comment (SilkeSchomann) [reply dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:66]: For me it looks like this:

image

File: dev-docs/source/Testing/MSlice/MSliceTestGuide.rst [hunk 4]
```diff
@@ -59,31 +106,187 @@ Taking Cuts
...
+
+#. Delete all workspaces apart from the ``MAR21335_Ei60meV`` workspace
+#. Scale the ``MAR21335_Ei60meV`` workspace with a factor of ``1.0``
+#. A new workspace with the name ``MAR21335_Ei60meV_scaled`` appears
+#. Select the ``MAR21335_Ei60meV`` workspace again, click on ``Add`` and select ``MAR21335_Ei60meV_scaled``, then click ``Ok``
+#. A new workspace with the name ``MAR21335_Ei60meV_sum`` appears
+#. Delete the workspace with the name ``MAR21335_Ei60meV_scaled``
+#. Scale the ``MAR21335_Ei60meV`` workspace with a factor of ``2.0``
+#. A new workspace with the name ``MAR21335_Ei60meV_scaled`` appears
+#. In the ``Workspace Manager`` tab select the workspace ``MAR21335_Ei60me_scaled``
```
Comment (sf1919) [inline dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:281]:
```suggestion
#. In the ``Workspace Manager`` tab select the workspace ``MAR21335_Ei60me_scaled``
```
```suggestion
#. In the ``Workspace Manager`` tab select the workspace ``MAR21335_Ei60meV_scaled``
```

Comment (sf1919) [inline dev-docs/source/Testing/MSlice/MSliceTestGuide.rst:281]:
```suggestion
#. In the ``Workspace Manager`` tab select the workspace ``MAR21335_Ei60me_scaled``
```
```suggestion
#. In the ``Workspace Manager`` tab select the workspace ``MAR21335_Ei60meV_scaled``
```

**Review by sf1919 — APPROVED** [review]
I've tested this not only by building the document but also by running through the tests themselves. There were only a few minor issues with the testing documentation and these have now been resolved. The documentation is clear and well illustrated and I am sure that when we come to testing MSlice at release this will be a huge asset.

Whilst testing I did come across some MSlice issues but these have either already been resolved and not yet merged into Mantid or they are being dealt with separately.

I am very happy to approve this PR.

**Pull request merged**